Libraries/Connect/Internal/Interceptors/ConnectInterceptor.swift

```diff
@@ -83,28 +83,51 @@ extension ConnectInterceptor: UnaryInterceptor {
                 ] = current.value
             })
 
-        if let encoding = response.headers[HeaderConstants.contentEncoding]?.first,
-           let compressionPool = self.config.responseCompressionPool(forName: encoding),
-           let message = response.message.flatMap({ try? compressionPool.decompress(data: $0) })
+        let finalResponse: HTTPResponse
+        let contentType = response.headers[HeaderConstants.contentType]?.first ?? ""
+        if response.code == .ok && !contentType.hasPrefix("application/\(self.config.codec.name())")
         {
-            proceed(HTTPResponse(
-                code: response.code,
-                headers: headers,
-                message: message,
-                trailers: trailers,
-                error: response.error,
+            // If content-type looks like it could be an RPC server's response, consider
+            // this an internal error.
+            let code: Code = contentType.hasPrefix("application/") ? .internalError : .unknown
```

Comment:
Is a non-200 HTTP status handled somewhere else? I expected to see it handled just above, or the HTTP status incorporated here (so deriving a code from the HTTP status instead of hard-coding to "unknown").

Reply:
The non-200 cases are falling into one of the `else`/`else if` cases below depending on whether the error response is compressed (`Connect Compressed Error and End-Stream/HTTPVersion:2/TLS:false/error/compressed` is one of these cases). Both of these use `response.code`, which is the Connect code from the HTTP status.